app/src/main/cpp/skyline/gpu.cpp

```diff
@@ -36,142 +37,38 @@ namespace skyline::gpu {
             ANativeWindow_unlockAndPost(window);
-            bufferQueue.FreeBuffer(buffer->slot);
+
             vsyncEvent->Signal();
-            if (prevTime != 0) {
-                auto now = std::chrono::duration_cast<std::chrono::microseconds>(std::chrono::steady_clock::now().time_since_epoch()).count();
-                state.logger->Error("{} ms, {} FPS", (now - prevTime) / 1000, 1000000 / (now - prevTime));
-            }
-            prevTime = std::chrono::duration_cast<std::chrono::microseconds>(std::chrono::steady_clock::now().time_since_epoch()).count();
-        }
-    }
+            texture->releaseCallback();
 
-    u32 GPU::OpenDevice(const std::string &path) {
-        state.logger->Debug("Opening NVDRV device ({}): {}", fdIndex, path);
-        auto type = device::nvDeviceMap.at(path);
-        for (const auto &device : fdMap) {
-            if (device.second->deviceType == type) {
-                device.second->refCount++;
-                fdMap[fdIndex] = device.second;
-                return fdIndex++;
-            }
-        }
-        std::shared_ptr<device::NvDevice> object;
-        switch (type) {
-            case (device::NvDeviceType::nvhost_ctrl):
-                object = std::make_shared<device::NvHostCtrl>(state);
-                break;
-            case (device::NvDeviceType::nvhost_gpu):
-            case (device::NvDeviceType::nvhost_vic):
-            case (device::NvDeviceType::nvhost_nvdec):
-                object = std::make_shared<device::NvHostChannel>(state, type);
-                break;
-            case (device::NvDeviceType::nvhost_ctrl_gpu):
-                object = std::make_shared<device::NvHostCtrlGpu>(state);
-                break;
-            case (device::NvDeviceType::nvmap):
-                object = std::make_shared<device::NvMap>(state);
-                break;
-            case (device::NvDeviceType::nvhost_as_gpu):
-                object = std::make_shared<device::NvHostAsGpu>(state);
-                break;
-            default:
-                throw exception("Cannot find NVDRV device");
-        }
-        deviceMap[type] = object;
-        fdMap[fdIndex] = object;
-        return fdIndex++;
-    }
+            if (frameTimestamp) {
+                auto now = util::GetTimeNs();
 
-    void GPU::CloseDevice(u32 fd) {
-        state.logger->Debug("Closing NVDRV device ({})", fd);
-        try {
-            auto device = fdMap.at(fd);
-            if (!--device->refCount)
-                deviceMap.erase(device->deviceType);
-            fdMap.erase(fd);
-        } catch (const std::out_of_range &) {
-            state.logger->Warn("Trying to close non-existent FD");
-        }
-    }
+                frametime = static_cast<u32>((now - frameTimestamp) / 10000); // frametime / 100 is the real ms value, this is to retain the first two decimals
+                fps = static_cast<u16>(1000000000 / (now - frameTimestamp));
```

Comment:
Might wanna make this a common constant, it's used for getTimeNs too

Reply:
I feel that might be a bit too generic to put in constants, thoughts ?

Reply:
Not really, if it's a constant it's a constant, not Ile there are any perf decreases with xonstexpr.